src/main/ApplicationImpl.cpp

```diff
@@ -198,6 +206,10 @@ ApplicationImpl::getNetworkID() const
 ApplicationImpl::~ApplicationImpl()
 {
     LOG(INFO) << "Application destructing";
+    if (mNtpSynchronizationChecker)
+    {
+        mNtpSynchronizationChecker->shutdown();
```

Comment:
Does this need to be called in the destructor? It is also called in gracefulStop   seems like it should just be called in one or the other